src/inference/hmc.jl

```diff
@@ -128,22 +128,21 @@ function AbstractMCMC.sample_init!(
         link!(spl.state.vi, spl)
         model(spl.state.vi, spl)
         theta = spl.state.vi[spl]
-        spl.state.z.θ .= theta
+        update_hamiltonian!(spl, model, length(theta))
+        spl.state.z = AHMC.phasepoint(rng, theta, spl.state.h)
```

Comment:
Consider adding a note here, explaining that we need this to refresh the internal cache phase point `z`'s hamiltonian energy.

Reply:
👍